tests/scripts/all.sh

```diff
@@ -1024,6 +1024,43 @@ component_check_test_cases () {
     unset opt
 }
 
+component_check_test_dependencies () {
+    msg "Check: test case dependencies: legacy vs PSA" # < 1s
+
+    found="check-test-deps-found-$$"
+    expected="check-test-deps-expected-$$"
+
+    # Find legacy dependencies in PSA tests
+    grep 'depends_on' tests/suites/test_suite_psa* |
+        grep -Eo '!?MBEDTLS_[^: ]*' |
+        grep -v MBEDTLS_PSA_ |
+        sort -u > $found
+
+    # Expected ones with justification - keep in sorted order!
```

Comment:
Thanks for the detailed explanations for each exception!

Please also add some text explaining the problem we want to solve. For someone who isn't familiar with the history here, I don't think that's easy to understand.